homeassistant/components/media_player/__init__.py

```diff
@@ -866,6 +866,14 @@ def preload_media_image_url(self, url):
             _async_fetch_image(self.hass, url), self.hass.loop
         ).result()
 
+    def set_last_bad_image_url(self, url):
+        """Save last bad image url."""
+        should_update = self._last_bad_image_url != url
+        self._last_bad_image_url = url
+        if should_update:
+            _LOGGER.debug('%s marked as inaccessible', url)
+            self.schedule_update_ha_state()
```

Comment:
If I understand what is happening correctly:

Without this call the effect would happen only the next time entity_picture is calculated. This will happen at unknown point in the future.

With this call the attribute will update and the frontend will update accordingly.

(Without this call the test fail, but I do see the blank image disappear at the UI - not sure why)

Is there a better call to be added from async context?